crates/rattler_index/src/lib.rs

```diff
@@ -63,12 +75,21 @@ pub fn package_record_from_index_json<T: Read>(
 /// and extract the package record from it.
 pub fn package_record_from_tar_bz2(file: &Path) -> std::io::Result<PackageRecord> {
     let reader = fs::File::open(file)?;
+    package_record_from_tar_bz2_reader(reader)
+}
+
+/// Extract the package record from a `.tar.bz2` package file.
+/// This function will look for the `info/index.json` file in the conda package
+/// and extract the package record from it.
+pub fn package_record_from_tar_bz2_reader(reader: impl Read) -> std::io::Result<PackageRecord> {
+    let bytes = reader.bytes().collect::<Result<Vec<u8>, _>>()?;
+    let reader = Cursor::new(bytes.clone());
     let mut archive = read::stream_tar_bz2(reader);
```

Comment:
note that this duplicates the file content in-memory. We didn't really find an elegant solution since `stream_tar_bz2` consumes the reader. But we're also still on amateur rust level, so happy to implement any suggested improvement 😅